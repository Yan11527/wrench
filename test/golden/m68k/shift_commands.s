	.data

input_addr: 	.word 0x80
output_addr:	.word 0x84

	.text

_start:
    movea.l		input_addr, A6
	movea.l		(A6), A6

	movea.l		output_addr, A7
	movea.l     (A7), A7

    move.l      (A6), D5
    lsl.l       1, D5
    move.l      D5, (A7)

    move.l      (A6), D5
    lsr.l       1, D5
    move.l      D5, (A7)

    move.l      (A6), D5
    asl.l       1, D5
    move.l      D5, (A7)

    move.l      (A6), D5
    asr.l       1, D5
    move.l      D5, (A7)
    
end:
    halt