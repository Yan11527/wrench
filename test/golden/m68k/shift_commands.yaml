name: assert lsl, lsr, asl, asr commands([5, 512, 8, 20]) == [10, 256, 16, 10]
limit: 2000
memory_size: 0x1000
input_streams:
  0x80: [5,512,8,20]
  0x84: []
reports:
  - name: Check results
    slice: last
    filter:
      - state
    view: |
      numio[0x80]: {io:0x80:dec}
      numio[0x84]: {io:0x84:dec}
    assert: |
      numio[0x80]: [] >>> []
      numio[0x84]: [] >>> [10,256,16,10]
